src/Range.tsx

```diff
@@ -258,12 +266,30 @@ class Range extends React.Component<IProps> {
         )
       );
     } else if (e.key === 'Tab') {
-      this.setState({ draggedThumbIndex: -1 });
+      this.setState({ draggedThumbIndex: -1 }, () => {
+        // If key pressed when thumb was moving, fire onFinalChange
+        if (isChanged) {
+          this.fireOnFinalChange();
+        }
+      });
+    } else {
+      if (isChanged) {
+        this.fireOnFinalChange();
+      }
     }
   };
 
   onKeyUp = (e: React.KeyboardEvent) => {
-    this.setState({ draggedThumbIndex: -1 });
+    const { isChanged } = this.state;
+    console.log('keyup', 'isChanged', isChanged);
```

Comment:
```suggestion
```

Reply:
Oops. Usually, linters don't let me this into a repository.